apps/meteor/tests/end-to-end/api/livechat/14-units.ts

```diff
@@ -409,4 +413,428 @@ import { IS_EE } from '../../../e2e/config/constants';
 		});
 	});
+
+	const testDepartmentsInUnit = (unitId: string, unitName: string, numDepartments: number) => {
+		return request
+			.get(api(`livechat/units/${unitId}`))
+			.set(credentials)
+			.expect(200)
+			.expect((res) => {
+				expect(res.body).to.have.property('_id', unitId);
+				expect(res.body).to.have.property('name', unitName);
+				expect(res.body).to.have.property('numMonitors', 1);
+				expect(res.body).to.have.property('numDepartments', numDepartments);
+				expect(res.body).to.have.property('type', 'u');
+			});
+	};
+
+	const testDepartmentAncestors = (departmentId: string, departmentName: string, ancestor?: string | null) => {
+		return request
+			.get(api(`livechat/department/${departmentId}`))
+			.set(credentials)
+			.expect('Content-Type', 'application/json')
+			.expect(200)
+			.expect((res) => {
+				expect(res.body).to.have.property('success', true);
+				expect(res.body).to.have.property('department');
+				expect(res.body.department).to.have.property('_id', departmentId);
+				expect(res.body.department).to.have.property('name', departmentName);
+				if (ancestor) {
+					expect(res.body.department).to.have.property('parentId', ancestor);
+					expect(res.body.department).to.have.property('ancestors').that.is.an('array').with.lengthOf(1);
+					expect(res.body.department.ancestors[0]).to.equal(ancestor);
+				}
+
+				if (ancestor === null) {
+					expect(res.body.department).to.have.property('parentId').that.is.null;
+					expect(res.body.department).to.have.property('ancestors').that.is.null;
+				}
+
+				if (ancestor === undefined) {
+					expect(res.body.department).to.not.have.property('parentId');
+					expect(res.body.department).to.not.have.property('ancestors');
+				}
+			});
+	};
+
+	describe('[POST] livechat/department', () => {
+		let departmentId: string;
+		let monitor1: Awaited<ReturnType<typeof createUser>>;
+		let monitor1Credentials: Awaited<ReturnType<typeof login>>;
+		let monitor2: Awaited<ReturnType<typeof createUser>>;
+		let monitor2Credentials: Awaited<ReturnType<typeof login>>;
+		let unit: IOmnichannelBusinessUnit;
+
+		const createDepartmentInUnit = (userCredentials: Credentials, departmentName: string, unitId: string) => {
+			return request
+				.post(api('livechat/department'))
+				.set(userCredentials)
+				.send({
+					department: { name: departmentName, enabled: true, showOnOfflineForm: true, showOnRegistration: true, email: 'bla@bla' },
+					departmentUnit: { _id: unitId },
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.property('department').that.is.an('object');
+					expect(res.body.department).to.have.property('name', departmentName);
+					expect(res.body.department).to.have.property('type', 'd');
+					expect(res.body.department).to.have.property('_id');
+					departmentId = res.body.department._id;
+				});
+		};
```

Comment:
It would be better to move this to the helpers instead and return the departmentId.

Reply:
I'll use the `createDepartment` helper we already have, instead (and just add some new params to it). We should just update this existing helper function at some point so that params are passed through an object
Please tell me if you think I should add a new helper instead

Reply:
You can update the existing one if possible.